pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,847 @@
+                        [interval(i_l) - interval(i_0) * imath.expm1(
+                            (interval(v) + interval(i)*interval(r_s)) /
+                            interval(nNsVth)) - interval(g_sh) *
+                            (interval(v) + interval(i)*interval(r_s)) -
+                            interval(i) for v, i in zip(v_gold, i_gold)]
+
+                    # Make sure the computed interval is within the specified
+                    #  tolerance, report bad value in exception if not
+                    for v, i, current_sum_residual_interval in zip(
+                            v_gold, i_gold,
+                            current_sum_residual_interval_list):
+                        if current_sum_residual_interval not in tol_interval:
+                            # Don't include entire curve in exception message
+                            del device_dict["iv_curves"][-1]["v_gold"]
+                            del device_dict["iv_curves"][-1]["i_gold"]
+                            raise ValueError("Residual interval {} not in {}, \
+                                 for (V, I)=({}, {}) at {}".format(
+                                current_sum_residual_interval, tol_interval, v,
+                                i, [
+                                    float.fromhex(item)
+                                    for item in device_dict["iv_curves"][-1]
+                                ]))
+
+        # Append this I-V curve to the list for this device
+        gold_dataset["devices"].append(device_dict)
+
+    # Return the list of devices' gold data with floats in hex format
+    return gold_dataset
+
+
+def convert_gold_dataset(gold_dataset):
+    """
+    Convert gold dataset into machine-specific format.
+
+    From a gold dataset in the lossless, machine-interoperable format, create
+    and return a copy in a numpy-based machine-specific format for the current
+    architecture that is compatible with the pvlib API.
+
+    Parameters
+    ----------
+    gold_dataset : dict
+        See ``make_gold_dataset()``.
+
+    Returns
+    -------
+    gold_dataset_converted : list
+        The gold dataset with floats converted from hex format, and shunt (or
+        parallel) conductance converted to shunt resistance.
+    """
+
+    # Return a new list instead of clobbering existing (uses more memory)
+    gold_dataset_converted = copy.deepcopy(gold_dataset)
+
+    # Convert the relevant header items
+    # FUTURE Switch pvlib over to shunt conductance instead of resistance
+    gold_dataset_converted["units"]["r_sh"] = "Ohm"
+    del gold_dataset_converted["units"]["g_sh"]
+
+    # Convert the values
+    for device in gold_dataset_converted["devices"]:
+        for iv_curve in device["iv_curves"]:
+            # Load G, T, and 5 single-diode model parameters for each I-V curve
+            # Floats converted from lossless, machine-interoperable hex format
+
+            iv_curve["G"] = float.fromhex(iv_curve["G"])
```

Comment:
should we use the pvlib keys?  `effective_irradiance` instead of `G`, `temp_cell` instead of `T`, etc.